### PR TITLE
[Custom Amounts M4] Change custom amounts dialog to full screen

### LINKED ISSUE #10393: [Custom Amounts M4] Change custom amount from dialog to full screen

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #10393 

### Description

This PR transforms the custom amount dialog into a full-screen fragment. We made this change due to the considerable amount of information on the custom amount screen, which proved to be overwhelming for a dialog format, especially in landscape mode. The transition to a full-screen fragment enhances the user interface and ensures better usability across different orientations.

### Testing instructions

1. Navigate to the order creation screen
2. Click on "Add Custom Amount" row
3. Ensure the custom amount is a full screen fragment
4. Ensure the custom amount UI behaves properly in the landscape mode

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/1331230/9804fe2d-1968-407d-b267-d15c70457b35

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt [hunk 1]
```diff
@@ -43,7 +41,7 @@ import java.math.RoundingMode
 import javax.inject.Inject
 
 @AndroidEntryPoint
-class CustomAmountsDialog : PaymentsBaseDialogFragment(R.layout.dialog_custom_amounts) {
+class CustomAmountsDialog : BaseFragment(R.layout.dialog_custom_amounts) {
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt]: How about renaming the class if it's not a dialog anymore?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt [hunk 2]
```diff
@@ -165,9 +145,7 @@ class CustomAmountsDialog : PaymentsBaseDialogFragment(R.layout.dialog_custom_am
                     type = viewModel.viewState.customAmountUIModel.type
                 )
             )
-        }
-        binding.imageClose.setOnClickListener {
-            cancelDialog()
+            viewModel.triggerExitEvent()
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt]: 💡 I'm not sure if it makes sense to call VM to trigger special event and then collect it. The previous approach with self-canceling dialog looks simpler. How about calling `findNavController().navigateUp()` directly?

**Review by samiuelson — APPROVED** [review]
LGTM, @AnirudhBhat 

I left a few questions, let me know what you think 👍🏼

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt]: Good point - Fixed here: https://github.com/woocommerce/woocommerce-android/pull/10394/commits/06e6152d1ce7d191f691b7f25f749850418818f6

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt (hunk 2 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/customamounts/CustomAmountsDialog.kt]: Done : https://github.com/woocommerce/woocommerce-android/pull/10394/commits/4ae3773736ba4664127bdd0536b93a55d2cd2863

**Pull request merged**